all/generate_gateway.sh

```diff
@@ -0,0 +1,300 @@
+#!/bin/bash
+
+set -e
+
+HTTP_PORT=8080
+
+printUsage() {
+  echo "Generates a docker image of a grpc-gateway server for a given service proto."  
+  echo
+  echo "The container will listen on port ${HTTP_PORT} for HTTP traffic, and proxy requests"
+  echo "to the gRPC service. In addition, it will serve a Swagger definition of its API at"
+  echo "/swagger.json"
+  echo
+  echo "Usage: generate_gateway.sh -f <service.proto> -p service_package -s ServiceName"
+  echo
+  echo "Options:"
+  echo "-h, --help                 Show this message."
+  echo "-f, --file FILE            The proto file to generate a gateway for."
+  echo "-s, --service SERVICE      The name of the gRPC service."
+  echo "-a, --address ADDRESS      The DNS address of the gRPC service."
+  echo "-p, --port PORT            The port the gRPC service is listening on."
+  echo "-c, --container CONTAINER  The name of the docker container to generate."
+  echo "-i EXTRA_INCLUDES                Extra proto includes."
+}
+
+#### PARSE ARGUMENTS ####
+# Path to the proto file.
+FILE=""
+# Name of the service
+SERVICE=""
+# Name of the container
+CONTAINER=""
+# Address of the gRPC service
+GRPC_SERVICE_HOST=""
+GRPC_SERVICE_PORT=""
+# Extra proto includes
+EXTRA_INCLUDES=""
+
+while test $# -gt 0; do
+  case "$1" in
+    -h|--help)
+      printUsage
+      exit 0
+      ;;
+    -f|--file)
+      shift
+      if test $# -gt 0; then
+        FILE=$1
+      else
+        printUsage
+        exit 1
+      fi
+      shift
+      ;;
+    -s|--service)
+      shift
+      if test $# -gt 0; then
+        SERVICE=$1
+      else
+        printUsage
+        exit 1
+      fi
+      shift
+      ;;
+    -c|--container)
+      shift
+      if test $# -gt 0; then
+        CONTAINER=$1
+      else
+        printUsage
+        exit 1
+      fi
+      shift
+      ;;
+    -a|--address)
+      shift
+      if test $# -gt 0; then
+        GRPC_SERVICE_HOST=$1
+      else
+        printUsage
+        exit 1
+      fi
+      shift
+      ;;
+    -p|--port)
+      shift
+      if test $# -gt 0; then
+        GRPC_SERVICE_PORT=$1
+      else
+        printUsage
+        exit 1
+      fi
+      shift
+      ;;
+    -i)
+      if test $# -gt 0; then
+        EXTRA_INCLUDES=$1
+      else
+        printUsage
+        exit 1
+      fi
+      shift
+      ;;
+    *)
+      printUsage
+      exit 1
+      ;;
+  esac
+done
+
+#### VALIDATE INPUT ####
+if [[ -z $FILE ]]; then
+    echo "Error: You must specify the proto file name"
+    printUsage
+    exit 1
+fi
+
+if [[ -z $SERVICE ]]; then
+    echo "Error: You must specify the proto service name"
+    printUsage
+    exit 1
+fi
+
+if [[ -z $CONTAINER ]]; then
+    echo "Error: You must specify the Docker container name"
+    printUsage
+    exit 1
+fi
+
+if [[ -z $GRPC_SERVICE_HOST ]]; then
```

Comment:
I don't think this and GRPC_SERVICE_PORT should be set here. I think the must be configured through env variables, or via flags on the container's entrypoint. https://github.com/spf13/viper has a good configuration hierarchy, and the main.go method should ensure these are set and present.